crates/sui-framework/packages/sui-framework/sources/authenticator_state.move

```diff
@@ -355,6 +364,56 @@ fun get_active_jwks(self: &AuthenticatorState, ctx: &TxContext): vector<ActiveJw
     self.load_inner().active_jwks
 }
 
+/// Tracks the set of addresses allowed to act as a given sender.
+///
+/// An alias allows transactions signed by the alias address to act as the
+/// original address. For example, if address X sets an alias of address Y, then
+/// then a transaction signed by Y can set its sender address to X.
+public struct AddressAliases has key {
+    id: UID,
+    aliases: vec_set::VecSet<address>,
+}
+
+/// Internal key used for derivation of AddressAliases object addresses.
+public struct AliasKey(address) has copy, drop, store;
+
+/// Provides the initial set of address aliases for the sender address.
+///
+/// By default, an address is its own alias. However, the original address can
+/// be removed from the set of allowed aliases after initialization.
+entry fun init_aliases(authenticator_state: &mut AuthenticatorState, ctx: &TxContext) {
```

Comment:
Could you add comments on why these need to be `entry`? Also wondering if this should return the `AddressAliases` instead of directly transferring it. For instance, now if I want to add an alias to my address for the first time, I would have to send two transactions.